### ISSUE TITLE
Enhancements to the Kubernetes extension

### ISSUE THREAD

**Issue by loicmathieu** [quarkusio/quarkus#29350]
### Description

I recently try the Kubernetes extension to generates Kubernetes ressources and I have some feedback / enhacement proposals : 

- When specifing container ressources, there is a log that seems not necessary
```
Apply requests cpu to:test-kubernetes
```
- The Ingress is using port value instead of port name, the deployment specify a port name of `http`, it would be better to use it instead of the port value.
- When adding Smallrye Health, probes are added to the container spec with the following settings:
```yaml
            initialDelaySeconds: 0
            periodSeconds: 30
            successThreshold: 1
```
As `initialDelaySeconds` is at 0, the first probe will always fails, and as `periodSeconds` is at 30s this means te container will be detected as RUNNING after 30s which is a lot for a typical Quarkus application that can starts in less than 3s even in constrainted environments. I'll suggest configuring  `initialDelaySeconds` to 3s or 5s.
I also wonder why the periodSeconds has been set to 30s when the default is 10s.
- There is no functionality to generates an HorizontalPodAutoscaling, using HPA in deployments is very common, this would be a great addition if Quarkus would be able to generate one.

**Comment by iocanel** [comment]
Thanks @loicmathieu these are some pretty valid points, especially regarding probes.

**Comment by loicmathieu** [comment]
@iocanel I had a look at this and:

My first point seems to be an issue at the fabric8 client level, nothing we can do about it.

My second point is not realy an issue as anything is generated.

For my 3rd point I propose PR #29580

For my last point I openned a separate issue a few days ago.

So I think I can mark this one as fixed by #29580
